graylog2-server/src/main/java/org/graylog/plugins/views/search/filter/OrFilter.java

```diff
@@ -57,6 +57,11 @@ public Filter.Builder toGenericBuilder() {
         return toBuilder();
     }
 
+    @Override
+    public Filter withFilters(Set<Filter> filters) {
+        return toBuilder().filters(filters).build();
+    }
```

Comment:
I believe that this method can be moved to the Filter interface as default method, instead being repeated 4 times. You would probably just need to add "toBuilder()" method to the Filter interface in order to achieve it.

Reply:
Good catch! :+1: